### PR TITLE
Add Xiaomi Yeelight Remote Control (YLYK01YL) support

### PR DESCRIPTION
**Pull request by syssi**
# What does this implement/fix? 

Adds the Yeelight Remote Control (YLYK01YL) support.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Configuration change (this will require users to update their yaml configuration files to keep working)

**Related issue or feature (if applicable):** fixes <link to issue>

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#<esphome-docs PR number goes here>
  
# Test Environment

- [x] ESP32
- [ ] ESP8266
- [ ] Windows
- [ ] Mac OS
- [ ] Linux

## Example entry for `config.yaml`:

```yaml
# Example config.yaml

esp32_ble_tracker:
#  scan_parameters:
#    interval: 160ms
#    window: 160ms

xiaomi_ylyk01yl:
  mac_address: "A4:C1:38:6C:23:2D"
  last_button_pressed:
    name: "last button pressed"
  on_button_on:
    then:
      - logger.log: "Button on"
  on_button_off:
    then:
      - logger.log: "Button off"
  on_button_sun:
    then:
      - logger.log: "Button sun"
  on_button_m:
    then:
      - logger.log: "Button m"
  on_button_plus:
    then:
      - logger.log: "Button plus"
  on_button_minus:
    then:
      - logger.log: "Button minus"

```

# Explain your changes

Describe your changes here to communicate to the maintainers **why we should accept this pull request**.
Very important to fill if no issue linked

## Checklist:
  - [x] The code change is tested and works locally.
  - [ ] Tests have been added to verify that the new code works (under `tests/` folder).
  
If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

### CODE REVIEW

File: esphome/components/xiaomi_ylyk01yl/__init__.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+                    cv.GenerateID(CONF_TRIGGER_ID): cv.declare_id(
+                        OnButtonMinusTrigger),
+                }
+            ),
+        }
+    )
+    .extend(esp32_ble_tracker.ESP_BLE_DEVICE_SCHEMA)
+    .extend(cv.COMPONENT_SCHEMA)
+)
+
```
Comment (jesserockz) [inline esphome/components/xiaomi_ylyk01yl/__init__.py:108]:
```suggestion


```

**Review by jesserockz — APPROVED** [review]
I feel like there is a lot of unnecessary duplication with the trigger code, and it could potentially be merged into one trigger with an optional `keycode` config param and variable to filter each one, or just trigger on all if that is not set (Good for sending an event to HA and processing there).

But otherwise this just needs some lines added to a test file and a docs PR and then can be merged.

**Comment by aronsky** [comment]
Hi @syssi, YLKG08YL/YLKG07YL work very similarly to this (they are dimmers intended for Xiaomi ceiling lights, that actually transmit clicks and rotations over BLE, instead of physically dimming the light). I'm planning on making a PR to add support for them, but maybe the 3 of them should be added all together, due to their similarities? I'm open for collaboration on this.

**Comment by syssi** [comment]
@aronsky I'm aware of the dimmer and I suggest to implement the device in a separate PR. It's easier to review.

**Comment by syssi** [comment]
I will care about the pending tasks asap!

**Comment by syssi** [comment]
I will add the tests asap.

**Pull request closed without merging**